docs/design/storagenode-graceful-exit.md

```diff
@@ -0,0 +1,89 @@
+	- Continue the graceful exit and exceed the bandwidth allocation the SNO originally setup
+	- Wait until the SNO has enough available bandwidth
+	- Cancel the graceful exit
+- When a Storage Node wants to rejoin the network I want them to have that ability so that they do not need to generate a new node ID via POW, go through the node vetting process, and so they can utilize their reputation. 
+- When a Storage Node rejoins the network I want the satellite to keep track of that so that we can start the escrow process for that node over. 
+
+## Design Overview
+
+- The repair checker will be responsible for finding pieces that need to be transferred from a node who is gracefully exiting the network.
+- The repair checker needs to add those pieces to another queue.
+	- This queue is responsible for prioritizing the order of pieces the exiting node sends to other nodes.
+		- pieces should be prioritized based on how many other pieces are still on the network. Pieces for files that are closer to needing to be repaired should be transferred off the exiting before pieces for files that are still considered healthy.
+	- This queue is responsible for communicating with the exiting node and telling it what piece to send to what node
+	- This should be a unique queue for each exiting node
+- The satellite needs to validate the correctness of the data the exiting node transferred to the new node
+	- The exiting node will send the new node the hash it has for the piece. (that hash contains the uplinks signature). The new node will sign it and send it back to the exiting node. The exiting node will send it to the satellite to prove the data was transferred to the new node. The satellite will check the uplinks signature and the new nodes signature on the hash it receives from the exiting node.
+- We need to create a new operation for storage nodes transferring pieces to other storage nodes
+	- Receiving storage nodes should be uploaded to as if they were receiving any other upload (orders, etc).
+- We need to have the storage node keep track how its progress during graceful exit by calculating how much data it had when it started the graceful exit and keeping track of how much data it has deleted (data is deleted from the exiting node as it sends pieces to new nodes). This is an estimation since the storage node is likely holding more data than it will gracefully exit (garbage data). 
+- The satellite needs to notify that the graceful exit has been completed so that the exiting node can delete the garbage data it is holding. 
+
+## Open Questions
+
+- What happens if we are doing a repair and a graceful exit on the same segment?
+- What happens if a piece is deleted when the piece is in the process of being transferred from the exiting node to other node in the network?
```

Comment:
Option 1. Leave the transferred piece as garbage on the new node. The GC process will take care of it.
Option 2. Once the exiting node sends the proof of transfer to the satellite, the satellite can check if the segment of this piece is still in the pointer db. If it was deleted, the satellite can send a delete request to the new node.

Reply:
if there is not that much overhead, I would prefer to implement the second option.

Reply:
i added those options to the design doc